src/Components/Components/src/ComponentsMetrics.cs

```diff
@@ -0,0 +1,235 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+using System.Diagnostics.Metrics;
+using Microsoft.AspNetCore.Components;
+using Microsoft.AspNetCore.Http;
+
+namespace Microsoft.AspNetCore.Components;
+
+internal sealed class ComponentsMetrics : IDisposable
+{
+    public const string MeterName = "Microsoft.AspNetCore.Components";
+    private readonly Meter _meter;
+
+    private readonly Counter<long> _navigationCount;
+
+    private readonly Histogram<double> _eventDuration;
+    private readonly Counter<long> _eventException;
+
+    private readonly Histogram<double> _parametersDuration;
+    private readonly Counter<long> _parametersException;
+
+    private readonly Histogram<double> _batchDuration;
+    private readonly Counter<long> _batchException;
+
+    public bool IsNavigationEnabled => _navigationCount.Enabled;
+
+    public bool IsEventDurationEnabled => _eventDuration.Enabled;
+    public bool IsEventExceptionEnabled => _eventException.Enabled;
+
+    public bool IsParametersDurationEnabled => _parametersDuration.Enabled;
+    public bool IsParametersExceptionEnabled => _parametersException.Enabled;
+
+    public bool IsBatchDurationEnabled => _batchDuration.Enabled;
+    public bool IsBatchExceptionEnabled => _batchException.Enabled;
+
+    public ComponentsMetrics(IMeterFactory meterFactory)
+    {
+        Debug.Assert(meterFactory != null);
+
+        _meter = meterFactory.Create(MeterName);
+
+        _navigationCount = _meter.CreateCounter<long>(
+            "aspnetcore.components.navigation.count",
```

Comment:
According to OTel naming guidelines `count` only gets used for UpDownCounters (https://opentelemetry.io/docs/specs/semconv/general/naming/#use-count-instead-of-pluralization-for-updowncounters). I don't know the details on how Blazor routing compares to ASP.NET routing but the prior art from ASP.NET is https://opentelemetry.io/docs/specs/semconv/dotnet/dotnet-aspnetcore-metrics/#metric-aspnetcoreroutingmatch_attempts. Does the action being counted have a similar concept and naming to ASP.NET 'match attempts' or does Blazor do it differently?